mne_connectivity/spectral/time.py

```diff
@@ -458,137 +468,111 @@ def _spectral_connectivity(data, method, kernel, foi_idx,
+        # squeeze time dimension
+        out[i] = out[i].squeeze(axis=-1)
 
-    # define the function to compute in parallel
-    parallel, p_fun, n_jobs = parallel_func(
-        pairwise_pli, n_jobs=n_jobs, verbose=verbose, total=total)
+    return out
 
-    return parallel(p_fun(s, t) for s, t in zip(source_idx, target_idx))
 
+def _plv(s_xy):
+    s_xy = s_xy / np.abs(s_xy)
+    plv = np.abs(s_xy.mean(axis=-1, keepdims=True))
+    return plv
 
-def _wpli(w, kernel, foi_idx, source_idx, target_idx, n_jobs, verbose, total,
-          faverage, weights):
-    """Pairwise weighted phase-lag index.
 
-    Input signal w is of shape (n_epochs, n_chans, n_tapers, n_freqs,
-    n_times)."""
-    def pairwise_wpli(w_x, w_y):
-        s_xy = _compute_csd(w[:, w_y], w[:, w_x], weights)
-        s_xy = _smooth_spectra(s_xy, kernel)
-        con_num = np.abs(s_xy.imag.mean(axis=-1, keepdims=True))
-        con_den = np.mean(np.abs(s_xy.imag), axis=-1, keepdims=True)
-        out = con_num / con_den
-        # mean inside frequency sliding window (if needed)
-        if isinstance(foi_idx, np.ndarray) and faverage:
-            return _foi_average(out, foi_idx)
-        else:
-            return out
+def _ciplv(s_xy):
+    s_xy = s_xy / np.abs(s_xy)
+    rplv = np.abs(np.mean(np.real(s_xy), axis=-1, keepdims=True))
+    iplv = np.abs(np.mean(np.imag(s_xy), axis=-1, keepdims=True))
+    ciplv = iplv / (np.sqrt(1 - rplv ** 2))
+    return ciplv
 
-    # define the function to compute in parallel
-    parallel, p_fun, n_jobs = parallel_func(
-        pairwise_wpli, n_jobs=n_jobs, verbose=verbose, total=total)
 
-    return parallel(p_fun(s, t) for s, t in zip(source_idx, target_idx))
+def _pli(s_xy):
+    pli = np.abs(np.mean(np.sign(np.imag(s_xy)),
+                         axis=-1, keepdims=True))
+    return pli
 
 
-def _cs(w, kernel, foi_idx, source_idx, target_idx, n_jobs, verbose, total,
-        faverage, weights):
-    """Pairwise cross-spectra."""
-    def pairwise_cs(w_x, w_y):
-        out = _compute_csd(w[:, w_y], w[:, w_x], weights)
-        out = _smooth_spectra(out, kernel)
-        if isinstance(foi_idx, np.ndarray) and faverage:
-            return _foi_average(out, foi_idx)
-        else:
-            return out
+def _wpli(s_xy):
+    con_num = np.abs(s_xy.imag.mean(axis=-1, keepdims=True))
+    con_den = np.mean(np.abs(s_xy.imag), axis=-1, keepdims=True)
+    wpli = con_num / con_den
+    return wpli
+
+
+def _coh(s_xx, s_yy, s_xy):
+    con_num = np.abs(s_xy.mean(axis=-1, keepdims=True))
+    con_den = np.sqrt(s_xx.mean(axis=-1, keepdims=True) *
+                      s_yy.mean(axis=-1, keepdims=True))
+    coh = con_num / con_den
+    return coh
 
-    # define the function to compute in parallel
-    parallel, p_fun, n_jobs = parallel_func(
-        pairwise_cs, n_jobs=n_jobs, verbose=verbose, total=total)
 
-    return parallel(p_fun(s, t) for s, t in zip(source_idx, target_idx))
+def _cs(s_xy):
```

Comment:
Apparently, this line is not covered in the tests. I don't see it in the docstring either. What does `cs` stand for? If it's another connectivity function, can you add the relevant section in the docstring.